src/components/dar_table/DarTableActions.js

```diff
@@ -0,0 +1,132 @@
+import { isEmpty, filter, isNil, find } from 'lodash/fp';
+import { h, div, span, a } from 'react-hyperscript-helpers';
+import { useMemo } from 'react';
+import { Storage } from '../../libs/storage';
+import DarTableVoteButton from './DarTableVoteButton';
+import DarTableOpenButton from './DarTableOpenButton';
+import DarTableCancelButton from './DarTableCancelButton';
+
+export default function DarTableActions(props) {
+  const { updateLists, openConfirmation, history, electionInfo, consoleType, extraOptions, index, baseStyle } = props;
+  const { election, dar, votes, researcher = {}, dac = {} } = electionInfo;
+  const currentUser = Storage.getCurrentUser();
+  const currentUserId = useMemo(() => Storage.getCurrentUser.dacUserId, []);
+  const currentUserRoles = currentUser.roles;
+  const chairVote = find((role) => role.name === 'Chairperson' && role.dacId === dac.dacId)(currentUser.roles);
+  const isChair = !isEmpty(currentUserRoles) && !isEmpty(dac) && !isNil(chairVote);
+  //template type is used to initialize general visibility options for buttons
+  //extraOptions is an object that contains boolean values for more granular control. Will be applied after template
+  //EXAMPLE: You want to use the chair setup but would like to see researcher buttons.
+  //extraOptions.showResearcher = true will activate that feature
+  //Both are intended to control button visibility for non-guaranteed features (ex: Vote)
+  const templates = {
+    chair: {
+      showVote: true,
+      showCancelIcon: true,
+      showResearcher: false
+    },
+    manageAccess: {
+      showVote: false,
+      showCancelIcon: false,
+      showResearcher: true
+    }
+  };
+
+  //applied extraOptions to template for final visibilityOptions reference
+  let visibilityOptions = templates[consoleType];
```

Comment:
To get past the codacy warning, would lodash's `get(consoleType)(templates)` work just as well here? Also, not entirely sure this would bypass codacy as we are passing in a variable as the path param here.

Reply:
Not sure, the function fundamentally does the same thing (lodash goes a step further and executes recursively for nested objects). Either way it's a false positive.